extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/KinesisRecordSupplier.java

```diff
@@ -747,6 +702,40 @@ private String getSequenceNumberInternal(StreamPartition<String> partition, Stri
 
   }
 
+  @VisibleForTesting
+  List<byte[]> deaggregateKinesisRecord(Record kinesisRecord) throws InvalidProtocolBufferException {
```

Comment:
I moved this function outside of the PartitionResource class so I could test it.  Is there a way to test private class methods?

Reply:
Not really a good one. That's usually what `@VisibleForTesting` means: "I would have made this private, except I needed to call it in a unit test."